uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -286,22 +286,30 @@ private static function validateTOILRequest($params, $absenceType, $absencePerio
     if($params['request_type'] !== self::REQUEST_TYPE_TOIL) {
       return;
     }
-
-    self::validateTOILToAccrueIsAValidOptionValue($params);
+    $isCalculationUnitInHours = AbsenceType::isCalculationUnitInHours($params['type_id']);
```

Comment:
This static method always bothered me a bit and seeing how it is being used here gave me an idea:

- Make it an instance method, instead of static. In terms of performance, it will be the same, since currently the method already creates an AbsenceType instance anyway. The difference is that now this instantiation will happen outside the method.
- Note that this `validateTOILRequest` method receives an AbsenceType instance. We can use that, and pass it to the other toil validation methods, the same way as we are currently doing for `validateTOILPastDays()`

I quickly checked other places that are also using this static method and I think they can easily be updated to use an instance method instead. What do you think?

Reply:
Good idea